### ISSUE TITLE
Confusing error message when indexing a string with a string

### ISSUE THREAD

**Issue by djmitche** [taskcluster/json-e#318]
```yaml
context:
  extra:
    attributes: a string!
template:
  $if: extra.attributes["someattr"] == 'x'
  then: true
```
will fail in the Python implementation with an error like `InterpreterError: infix: [..] expects integer [..] integer`.  That's not very helpful!  Let's add a test and give a better error message, consistently across all supported implementation languages.

**Comment by contrepoint** [comment]
Hi @owlishDeveloper, may I work on this issue? Thank you!

**Comment by owlishDeveloper** [comment]
All yours!

**Comment by stuartelimu** [comment]
can i work on this issue ?

**Comment by owlishDeveloper** [comment]
Fixed in #324
